### ISSUE TITLE
Highlighting text with the mouse in split view causes the other view to move

### ISSUE THREAD

**Issue by reinux** [VsVim/VsVim#2664]
**Describe the bug**
In split view, highlighting a section of text in one view causes the other view to scroll to the same position, which defeats the purpose of split view.

**To Reproduce**

In VS2019:

1. Window -> Split
2. Scroll to some location in the code in one view different from the other
3. Select some text with the mouse

**Expected behavior**
The other view stays put.

**Screenshots**
https://imgur.com/j2YE8wQ

**Environment (please complete the following information):**

- Visual Studio version: VS2019 Community
- VsVim version: 2.7.0.0
- Programming Language: agnostic (F# in example, but also tested in C#, .ini files etc.)
- Check(Type 'x') any that are installed:
    - [ ] ReSharper
    - [ ] Visual Assist

**Comment by ricksladkey** [comment]
Thanks for reporting this problem; I can reproduce this bug.

I'll take a look at getting it fixed.

**Comment by ricksladkey** [comment]
Researching this problem has led me to an uncomfortable realization that having one mode for **all** text views associated with a text buffer is vim-compatible, but a seriously broken model for VsVim.

I would like to know if @jaredpar approves (conceptually) of my moving `IVimTextBuffer.ModeKind` up to `IVimBufferData.ModeKind`? This would allow two split windows into the same text buffer to be in different modes.

Right now it works "okay" for insert and normal mode, but for select and visual modes the fact that both text views have separate and distinct carets and selections while both being in visual mode is a broken scenario. In Vim they would have literally the same exact selection. For us to emulate that, we would have to detect that there are other text views for the same text buffer and adjust their selections to be identical, which is:

1. A lot of work
2. Of no real value

VsVim is ideally designed from the outset to cope with two text views into the same buffer being in different modes. In particular, every text view has its own caret and selection. Vim could not handle that situation well, and so the fact that all windows for the same underlying buffer are in the same mode is a really a limitation of Vim rather than a conscious decision.

To put it another way, fixing the bug identified in this issue is easy, but to fix the root problem we would have to sync the selection between all text views that share the same text buffer, and to me that seems like a bad idea.

**Comment by ricksladkey** [comment]
I know this is complicated, but here is a scenario to make it clear why the current model is broken:

- One document window is open
- The user splits the document into two windows
- The upper window moves the caret to the first line
- The second window moves the caret to the middle of the file
- The second window enters visual line mode
- This forces the underlying text buffer into line mode
- The first window is now in visual line mode for ... what lines?

It makes no sense that the upper window is in visual line mode for the first line and the lower window is in visual line mode for some line in the middle of the file. What would make sense is if the upper window were in normal mode with the caret on the first line and the lower (focused) window was in whatever mode the user put it in, including visual line mode.

Aw gee, I just realized undo transactions occur (partially) at the text buffer level. Are we at risk of corrupting the undo chain if we have independent modes for the same underlying text buffer?

**Comment by jaredpar** [comment]
> VsVim is ideally designed from the outset to cope with two text views into the same buffer being in different modes

Indeed. When I initially started VsVim I didn't realize that vim sync'd modes between different views. Hence the code was initially designed to support views being in different modes. Can't remember exactly when but probably a few years into the project before I realized that modes were sync'd. 

> I would like to know if @jaredpar approves (conceptually) of my moving IVimTextBuffer.ModeKind up to IVimBufferData.ModeKind? This would allow two split windows into the same text buffer to be in different modes.

Is there anything we lose here other than strict vim compatibility? I've run into problems before keeping modes sync'd and had similar thoughts about breaking them up. I'm unsure if this vim behavior provides value to users or if it's just a bit of trivia I implemented for strict compat state. IIRC it's actually very hard for customers to observe they are linked. 

There are also cases where where it's just not possible to keep them sync'd. Consider that every `IVimBuffer` starts in the `Uninitialized` mode due to hard requirements around WPF editor initialization. Existing `IVimTextBuffers` though don't go into `Uninitialized` mode when a new `IVimBuffer` is created. Hence the idea that they are always sync'd is already a bit of a broken model and it causes a deal of pain to maintain that. 

> Aw gee, I just realized undo transactions occur (partially) at the text buffer level. Are we at risk of corrupting the undo chain if we have independent modes for the same underlying text buffer?

Don't believe that is a problem. The only real issue with Undo / Redo is that the core editor will only update `ITextView` information for a single instance, not every `ITextView` that is connected to the `ITextBuffer`. 

Which is yet another place where the vim compatibility breaks down. An undo in vim will update caret positions in every view but in VsVim it only updates one view (#2674).
